test/gbdt/test_gbdt.py

```diff
@@ -59,3 +65,59 @@ def test_gbdt(gbdt_cls, stypes, task_type_and_metric):
         assert (0 <= score <= 1)
     elif task_type == TaskType.MULTICLASS_CLASSIFICATION:
         assert (0 <= score <= 1)
+
+
+@pytest.mark.parametrize('gbdt_cls', [
+    CatBoost,
+    XGBoost,
+    LightGBM,
+])
+@pytest.mark.parametrize('task_type_and_metric', [
+    (TaskType.REGRESSION, Metric.RMSE),
+    (TaskType.REGRESSION, Metric.MAE),
+    (TaskType.BINARY_CLASSIFICATION, Metric.ACCURACY),
+    (TaskType.BINARY_CLASSIFICATION, Metric.ROCAUC),
+    (TaskType.MULTICLASS_CLASSIFICATION, Metric.ACCURACY),
+])
+def test_save_load_gbdt(gbdt_cls, task_type_and_metric):
+    task_type, metric = task_type_and_metric
+    dataset: Dataset = FakeDataset(
+        num_rows=30,
+        with_nan=True,
+        stypes=[stype.numerical, stype.numerical, stype.text_embedded],
+        create_split=True,
+        task_type=task_type,
+        col_to_text_embedder_cfg=TextEmbedderConfig(
+            text_embedder=HashTextEmbedder(8)),
+    )
+    dataset.materialize()
+    gbdt = gbdt_cls(
+        task_type=task_type,
+        num_classes=dataset.num_classes
+        if task_type == TaskType.MULTICLASS_CLASSIFICATION else None,
+        metric=metric,
+    )
+    copy_gbdt = copy.deepcopy(gbdt)
+
+    path = osp.join(TEST_DIR.name, f'{gbdt_cls.__name__}.txt')
```

Comment:
why `.txt`? Is it saved in a text format?

Reply:
It is a binary file, but the [official doc](https://lightgbm.readthedocs.io/en/latest/Python-Intro.html#training) gives an example with `.txt`, so I referred to it.